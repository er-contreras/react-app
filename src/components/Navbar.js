import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './NavBar.module.css';

const Navbar = () => { // eslint-disable-line
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote',
      text: 'Quote',
    },
  ];

  return (
    <nav className={styles.navBar}>
      <div id={styles.title}>
        <h1>Math Magicians</h1>
      </div>
      <ul>
        {links.map((link) => { // eslint-disable-line
          return (
            <li key={link.id}>
              <NavLink to={link.path} activeClassName={styles.activelink}>
                {link.text}
              </NavLink>
            </li>
          );
        })}
      </ul>
    </nav>
  );
};

export default Navbar;
