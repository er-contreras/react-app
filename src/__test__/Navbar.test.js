import React from 'react';
import Navbar from '../components/Navbar';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

it('Should render same text passed into ...', () => {
  const tree = TestRenderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the main heading', () => {
  render(<Navbar />);
  const headingElement = screen.getByText(/^Math Magicians$/i);

  expect(headingElement).toBeInTheDocument();
});