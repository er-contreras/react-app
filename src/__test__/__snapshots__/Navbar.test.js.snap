// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test Should render same text passed into ... 1`] = `
<nav
  className="navBar"
>
  <div
    id="title"
  >
    <h1>
      Math Magicians
    </h1>
  </div>
  <ul>
    <li>
      <a
        aria-current={null}
        href="#/home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="#/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="#/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Navbar test Test Calculator 1`] = `
<a
  href="#/calculator"
>
  Calculator
</a>
`;

exports[`Navbar test Test Home 1`] = `
<a
  href="#/home"
>
  Home
</a>
`;

exports[`Navbar test Test Quote 1`] = `
<a
  href="#/quote"
>
  Quote
</a>
`;

exports[`Navbar test renders the main heading 1`] = `
<h1>
  Math Magicians
</h1>
`;
